src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java

```diff
@@ -52,15 +52,20 @@ public void deploy(DeployConfiguration config) throws AppEngineException {
     Preconditions.checkNotNull(config);
     Preconditions.checkNotNull(config.getDeployables());
     Preconditions.checkArgument(config.getDeployables().size() > 0);
+    File workingDirectory = null;
 
     List<String> arguments = new ArrayList<>();
     arguments.add("deploy");
-    for (File deployable : config.getDeployables()) {
-      if (!deployable.exists()) {
-        throw new IllegalArgumentException(
-            "Deployable " + deployable.toPath() + " does not exist.");
+    if (config.getDeployables().size() == 1 && config.getDeployables().get(0).isDirectory()) {
+      workingDirectory = config.getDeployables().get(0);
+    } else {
+      for (File deployable : config.getDeployables()) {
+        if (!deployable.exists()) {
+          throw new IllegalArgumentException(
```

Comment:
I hesitate changing this right now because throwing "IllegalArgumentException when a local deployable referenced by the configuration isn't found" is the behavior that existed for a while.